gwpy/plot/plot.py

```diff
@@ -521,9 +521,9 @@ def add_segments_bar(self, segments, ax=None, height=0.14, pad=0.1,
             pass
 
         # add new axes
-        if ax.get_axes_locator():
+        try:
             divider = ax.get_axes_locator()._axes_divider
-        else:
+        except (AttributeError, TypeError):
```

Comment:
what's the reason for handling both of these errors? diff versions of matplotlib use different errors?

Reply:
This has been simplified.